scipy/interpolate/tests/test_bsplines.py

```diff
@@ -324,6 +343,13 @@ def test_integral(self):
         assert_allclose(b.integrate(-1, 1), 0)
         assert_allclose(b.integrate(-1, 1, extrapolate=True), 0)
         assert_allclose(b.integrate(-1, 1, extrapolate=False), 0.5)
+        assert_allclose(b.integrate(-1, 1, extrapolate='periodic'), 1)
+        assert_allclose(b.integrate(1, -1, extrapolate='periodic'), -1)
```

Comment:
To be honest I feel these two tests are weak. There are several cases for periodic integration, ideally they should be all covered. See `test_integrate_periodic` in `test_interpolate` for some reference. And try to cover more or less all possible cases.